File: 0169-majority-element/0169-majority-element.cpp
// class Solution {
// public:
//     int majorityElement(vector<int>& nums) {
//         unordered_map<int,int> m;
//         for(int i=0;i<nums.size();i++)
//             m[nums[i]]++;
//         int res=0,maxi=0;
//         for(int i=0;i<nums.size();i++)
//         {   if(m[nums[i]]>maxi)
//                {
//                    maxi=m[nums[i]];
//                    res=nums[i];
//                 }
               
//                }
// return res;
//                }
// };

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        return nums[nums.size()/2];
    }
};

File: 0169-majority-element/0169-majority-element.java
// class Solution {
// public:
// //     int majorityElement(vector<int>& nums) {
// //         // sort(nums.begin(), nums.end());
// //         // return nums[nums.size()/2];
// //         int major=num[0], count = 1;
// //         for(int i=1; i<num.length;i++){
// //             if(count==0){
// //                 count++;
// //                 major=num[i];
// //             }else if(major==num[i]){
// //                 count++;
// //             }else count--;
            
// //         }
// //         return major;
// //     }
    
//     int major=num[0], count = 1;
//         for(int i=1; i<num.length;i++){
//             if(count==0){
//                 count++;
//                 major=num[i];
//             }else if(major==num[i]){
//                 count++;
//             }else count--;
            
//         }
//         return major;
// }

public class Solution {
    public int majorityElement(int[] num) {

        int major=num[0], count = 1;
        for(int i=1; i<num.length;i++){
            if(count==0){
                count++;
                major=num[i];
            }else if(major==num[i]){
                count++;
            }else count--;
            
        }
        return major;
    }
}